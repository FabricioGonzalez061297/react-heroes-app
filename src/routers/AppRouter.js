import React, { useContext } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { AuthContext } from '../auth/AuthContext';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoutes } from './PrivateRoutes';
import { PublicRoutes } from './PublicRoutes';

export const AppRouter = () => {

    const {user:{logged:isAuthenticated} } = useContext(AuthContext);

    return (
       <Router>
           <div>

                <Switch>
                    <PublicRoutes  isAuthenticated={isAuthenticated}  exact path="/login" component={LoginScreen} />
                    <PrivateRoutes isAuthenticated={isAuthenticated} path="/" component={DashboardRoutes} />
                </Switch>
           </div>
       </Router>
    )
}
