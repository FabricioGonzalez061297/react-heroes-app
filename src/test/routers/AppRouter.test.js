import { mount } from "enzyme";
import React from 'react'
import { AuthContext } from "../../auth/AuthContext";
import { AppRouter } from "../../routers/AppRouter";


describe('pruebas en <AppRouter/>', () => {

        const authContext={
            user:{logged:false},
            dispatch:jest.fn()
        }

        test('debe de mostrar login sino esta autenticado', () => {
            const wrapper=mount(
                <AuthContext.Provider value={authContext} >
                    <AppRouter/>
                </AuthContext.Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('Debe de mostrar el componente inicial si esta autenticado', () => {

            
            const authContext={
                user:{logged:true,name:'fabricio'},
                dispatch:jest.fn()
            }
            const wrapper=mount(
                <AuthContext.Provider value={authContext} >
                    <AppRouter/>
                </AuthContext.Provider>
            );

            expect(wrapper.find('.navbar').exists()).toBe(true);

        });
        
        
})
