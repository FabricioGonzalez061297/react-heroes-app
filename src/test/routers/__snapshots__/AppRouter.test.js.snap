// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <AppRouter/> debe de mostrar login sino esta autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <form>
      <div
        className="imgcontainer"
      >
        <img
          alt="user"
          className="avatar"
          src="https://img.icons8.com/bubbles/2x/user-male.png"
        />
      </div>
      <div
        className="container_app"
      >
        <label
          htmlFor="uname"
        >
           
          <b>
            Usuario
          </b>
        </label>
        <input
          name="uname"
          placeholder="Ingrese su usuario"
          type="text"
        />
        <label
          htmlFor="pwd"
        >
           
          <b>
            Contraseña
          </b>
           
        </label>
        <input
          name="pwd"
          placeholder="Ingrese su contraseña"
          type="password"
        />
        <button
          className="login"
          onClick={[Function]}
          type="button"
        >
            Ingresar
        </button>
        <label
          style={
            Object {
              "marginBottom": "20px",
            }
          }
        >
           
          <input
            name="remenber"
            type="checkbox"
          />
           Recordarme 
        </label>
      </div>
      <div />
    </form>
  </div>
</div>
`;
